### PR TITLE
Cc feature/30/add tag to a post

### LINKED ISSUE #29: Remove a Tag from a Post
As an author I would like to be able to unassociate one or more Tags from one of my Posts so that I can correct a mistakenly added Tag.

**Given** the author is viewing Post details  
**When** they select the `Manage Tags` option  
**Then** they should be presented with a selection of Tags options that are associated with the Post  

**Given** the author has unchecked the Tags they wish to remove from a Post  
**When** they click the `Save` button  
**Then** the association should be removed from the database  
**And** the user should be directed to the Post details page  
**And** the removed Tags should no longer be displayed on the Post details page

### LINKED ISSUE #30: Add a Tag to a Post
As an author I would like to be able to associate one or more Tags with one of my Posts so that readers can easily see the Post topics at a glance.

**Given** the author is viewing Post details  
**When** they select the `Manage Tags` option  
**Then** they should be presented with a selection of Tags options to associate with the Post  

**Given** the author has selected the Tags they wish to associate with a Post  
**When** they click the `Save` button  
**Then** the association should be saved to the database  
**And** the user should be directed to the Post details page  
**And** the selected Tags should be displayed on the Post details page

### PR DESCRIPTION
**Pull request by ccote95**
### Purpose
To implement a modal for the user to assign/unassign tags to a post

### Files Changed
- Created a file to hold the PostTagsModal in `client/src/components/modals/PostTagsModal.jsx`
- Created a modal to render a checkbox for each tag, rendering the tags associated with the post checked in `client/src/components/modals/PostTagsModal.jsx`
- Implemented logic to pass to the PostTagsModal in `client/src/components/posts/PostDetails.jsx`
- Created a controller for postTags in `Controllers/PostTagController.cs`
- Created a manager to send the new tags to the controller in `client/src/managers/postTagManager.js`

### Testing
Fetch, setup, run according to the projects README. but do not create a migration.
Confirm the following:
- As an author, when viewing one of your posts a manage tags button should be rendered
- When a author clicks the manage tags button a modal should render with checkboxes for each tag in the database
- The tags that are currently assigned to the post should render checked
- After the author has chosen to assign more or less tags to the post, when you click save the modal will close, rendering the new tags on the post

closes #29 
closes #30

### CODE REVIEW

**Review by Michael-Deitz — APPROVED** [review]

**Review by joshuahamrick — APPROVED** [review]

File: Controllers/PostTagController.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            _dbContext.Add(new PostTag()
+            {
+                PostId = postId,
+                TagId = newTagId
+
+            }
+            );
+        }
+
+        // tagIds.Select(t => _dbContext.PostTags.Add(
```
Comment (ztrouy) [inline Controllers/PostTagController.cs:44]: Remove commented out code

File: client/src/components/modals/PostTagsModal.jsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+    const handleSave = () => {
+        update(post.id,selectedTagIds).then(() => {refresh()});
+       
+        toggleModal();
+    }
+
+    return(
+        <Modal isOpen={isModalOpen}>
+        <ModalHeader toggle={toggleModal}>
```
Comment (ztrouy) [inline client/src/components/modals/PostTagsModal.jsx:42]: Clean up the formatting/indentation for this modal

File: client/src/components/posts/PostDetails.jsx [hunk 3]
```diff
@@ -2,18 +2,32 @@ import { useEffect, useState } from "react"
 import { useParams } from "react-router-dom"
 import { getApprovedAndPublishedPostById } from "../../managers/postManager.js"
 import PageContainer from "../PageContainer.jsx"
-import { Badge, Button, Card, CardBody, CardImg, CardImgOverlay, CardSubtitle, CardText, CardTitle, Spinner } from "reactstrap"
+import { Badge, Button, Card, CardBody, CardImg, CardImgOverlay, CardSubtitle, CardText, CardTitle, FormGroup, Input, Label, Modal, ModalBody, ModalHeader, Spinner } from "reactstrap"
```
Comment (ztrouy) [inline client/src/components/posts/PostDetails.jsx:5]: Remove unused imports

File: client/src/components/posts/PostDetails.jsx [hunk 4]
```diff
@@ -138,6 +156,12 @@ export const PostDetails = ({ loggedInUser }) => {
                     </div>
                 </CardBody>
             </Card>
+            <PostTagsModal
+            refresh = {refresh}
```
Comment (ztrouy) [inline client/src/components/posts/PostDetails.jsx:160]: Indent props

**Review by ztrouy — CHANGES_REQUESTED** [review]

**Review by ztrouy — APPROVED** [review]

**Pull request merged**